### PR TITLE
adapt your number if you already have history files

### PR DESCRIPTION
**Pull request by muthmann**
You can now stop crawling for historical data and continue later on.
Also added a print with the date of every data-request, to get an overview about the process

### CODE REVIEW

File: history_crawler.py [hunk 1]
```diff
@@ -78,40 +80,44 @@
...
         url = URL + "&from=" + str(start)
-    elif exchange in ['binance_future','binanceSpot']:
-        url= URL + "&startTime="+str(start)
+    elif exchange in ['binance_future', 'binanceSpot']:
+        url = URL + "&startTime="+str(start)
     elif exchange == 'phemex':
         url = URL + "&from=" + str(start)+"&to="+str(start+2000*60)
 
+    parsed_url = urlparse(url)
+    date_str = parse_qs(parsed_url.query)['from'][0]
```
Comment (kuegi) [inline history_crawler.py]: this only works for phemex and bybit. and date_str is always equal to the value of start, so no need to add the urlparse to get it.

Comment (kuegi) [reply history_crawler.py]: if you want to print the current timestamp of the crawling, better print it within the if - elif. cause there you know: what value is used and how this value is converted to a date. for some exchanges you can't convert it to a date, as its just a sequence-number.
if you want to print the tstamp of the most recent bar in the history, why not take it from result[-1].tstamp ?

**Review by kuegi — COMMENTED** [review]
Hi and thx for the PR. i am just curious: do you use the bot in production already? are you already in the discord-server?
(or do i already know you and just don't recognize the handle? ;)

File: history_crawler.py (hunk 1 above)
Comment (muthmann) [reply history_crawler.py]: Thanks for your answer, made some new changes.
I have applied the time logging for all exchanges. While that I have seen that there were some problems with the api's, made some changes on that too (maybe the exchanges have changed a few things on that).

File: history_crawler.py [hunk 2]
```diff
@@ -7,16 +7,28 @@
...
+    else:
+        symbol = 'BTCUSDT'
+elif exchange == 'bitmex':
+    if len(sys.argv) > 2 and sys.argv[2][len(sys.argv[2])-1] == 'T':
+        symbol = sys.argv[2]
+    else:
+        symbol = 'XBT'
+else:
+    symbol = sys.argv[2] if len(sys.argv) > 2 else 'BTCUSD'
+print(f'crywling from: {exchange}, with symbol: {symbol}')
```
Comment (kuegi) [inline history_crawler.py]: crywling sometimes is appropriate, but i guess you meant "crawling"

**Comment by kuegi** [comment]
@MaimP better remove your api key and secrest ;)

**Comment by kuegi** [comment]
@MaimP history is still there. you need to amend or reset your local commits and force push (see discord)

**Pull request closed without merging**